kafka-rest/src/main/java/io/confluent/kafkarest/controllers/SchemaManagerImpl.java

```diff
@@ -31,47 +33,101 @@
 
 final class SchemaManagerImpl implements SchemaManager {
   private final SchemaRegistryClient schemaRegistryClient;
-  private final SubjectNameStrategy subjectNameStrategy;
 
   @Inject
-  SchemaManagerImpl(
-      SchemaRegistryClient schemaRegistryClient, SubjectNameStrategy subjectNameStrategy) {
+  SchemaManagerImpl(SchemaRegistryClient schemaRegistryClient) {
     this.schemaRegistryClient = requireNonNull(schemaRegistryClient);
-    this.subjectNameStrategy = requireNonNull(subjectNameStrategy);
   }
 
   @Override
-  public RegisteredSchema getSchemaById(EmbeddedFormat format, int schemaId) {
-    checkArgument(format.requiresSchema(), "%s does not support schemas.", format);
+  public RegisteredSchema getSchemaById(String subject, int schemaId) {
+    ParsedSchema schema;
+    try {
+      schema = schemaRegistryClient.getSchemaBySubjectAndId(subject, schemaId);
+    } catch (IOException | RestClientException e) {
+      throw new SerializationException("Error when fetching schema by id.", e);
+    }
+
+    int schemaVersion;
+    try {
+      schemaVersion = schemaRegistryClient.getVersion(subject, schema);
+    } catch (IOException | RestClientException e) {
+      throw new SerializationException("Error when fetching schema version.", e);
+    }
+
+    return RegisteredSchema.create(subject, schemaId, schemaVersion, schema);
+  }
 
+  @Override
+  public RegisteredSchema getSchemaById(
+      SubjectNameStrategy subjectNameStrategy, String topicName, boolean isKey, int schemaId) {
```

Comment:
if all the parameters really are needed, maybe rename to just `getSchema`.

Reply:
Completely rewrote this bit. Ack.